frontend/app/src/components/defi/wizard/DefiWizard.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <v-card>
+    <v-stepper v-model="step" vertical>
+      <v-stepper-step :complete="step > 1" step="1">
+        Setup DeFi
+      </v-stepper-step>
+      <v-stepper-content step="1">
+        <v-card class="mb-12" height="200px" outlined color="grey lighten-4">
+          <v-card-title>
+            Select options to improve your Rotki DeFi experience
+          </v-card-title>
+          <v-card-text>
+            <p>
+              Rotki will attempt by default to retrieve information for all your
```

Comment:
By default, rotki will attempt to retrieve information on each DeFi protocol for all of your accounts. This retrieval may require a considerable amount of time depending on the number of accounts you track.

Reply:
> Finally. No accounts selected for a protocol should end up with the default of querying all accounts for that module.
> 
> **Important**: But ... having no modules/protocols set should not default to all of them. It should actually default to "no protocols" set. Someone may want to disable DeFi completely.

I think we can benefit by just being more explicit in each button with respect to what it does.

In stage 1 have three "buttons": 

(1) I want to enable all defi modules for all accounts
(autofills in the necessary info and makes the backend call)

(2) I want to select which defi modules and which accounts I would like to enable
(takes you to stage 2)

(3) I want to disable all defi modules
(autofills in the necessary info and makes the backend call)

Another important thing we need to remember is that the user might not know at the beginning of the wizard which accounts "have defi", or how much each account has. In general we are introducing a lot of complexity here to solve a certain problem (slow delivery of defi details due to how we're getting them) -- which is on our end -- to the user. We should strive to simplify the process as much as possible until eventually we get a defi querying mechanism (whether via quickblocks / the graph / whatever) that makes this process a fallback mechanism rather than the "main route". 

Can we set up / offer something like the below?
* At the start of the wizard (or ideally when an account is entered) -- run all defi modules + accounts in the background, identifying which accounts have "defi balances" so we can identify for the user which accounts those are. This will make the process of selecting which accounts a user wants to enable drastically easier on their end.
* Allow for the option to do this background processing automatically every X hours (e.g. 24 hrs) and then update the list of "enabled protocols and accounts" automatically in the backend. For example: we poll all accounts/addresses once per X hours, flag which accounts have defi activity with which protocols, and enable only those until (a) the threshold has passed or (b) the user does a hard refresh or (c) a manual override via the defi wizard.

Reply:
> At the start of the wizard (or ideally when an account is entered) -- run all defi modules + accounts in the background, identifying which accounts have "defi balances" so we can identify for the user which accounts those are. This will make the process of selecting which accounts a user wants to enable drastically easier on their end.

So this is not so much for the DeFi balances, this is not slow even with 30 accounts. It's for the historical DeFi data of a protocol. And you can still have 0 balance in a protocol but have extensively used it in the past. So such an approach may be misleading. I thought of doing this in the past but figured it's not a good idea as it's misleading. Even in my accounts I have abandoned protocols for which I would still like to see historical data.

> Allow for the option to do this background processing automatically every X hours (e.g. 24 hrs) and then update the list of "enabled protocols and accounts" automatically in the backend. For example: we poll all accounts/addresses once per X hours, flag which accounts have defi activity with which protocols, and enable only those until (a) the threshold has passed or (b) the user does a hard refresh or (c) a manual override via the defi wizard.

That is doable in the backend eventually. Not easy .. as there is A LOT of such backend calls that should run "on their own time". But we could run such a call in the backend. But then the question is also how to show to the user if there is a difference between what he has enabled and what we found out.

Also a user may not want to see historical data for a protocol for some reason. For example I no longer care about DSR data. I want to disable it even though 2 of my accounts did participate. I want faster queries.

Reply:
> So this is not so much for the DeFi balances, this is not slow even with 30 accounts. It's for the historical DeFi data of a protocol. And you can still have 0 balance in a protocol but have extensively used it in the past. So such an approach may be misleading. I thought of doing this in the past but figured it's not a good idea as it's misleading. Even in my accounts I have abandoned protocols for which I would still like to see historical data.

I get your point, but we still somehow need to address the issue of us (the app) expecting the user to know what's in an account without giving them any information. Perhaps me writing "balances" was misleading here, since the suggestion was that we will do a "preload" of all protocols for all accounts, we can indicate whether the account has "defi history" (and/or balances) or not.

I didn't have the chance to load the PR locally but this is another important thing that we need to point out (maybe kelsos already has) in Stage 1: Disabling the modules should not not disable them in the "overview", only for "details" or "histories". Since overview balances are indeed fast we need to double-check that this is the case (to test: Aave & Maker overviews work properly and show same "completeness" of info with modules disabled).

> That is doable in the backend eventually. Not easy .. as there is A LOT of such backend calls that should run "on their own time". But we could run such a call in the backend. But then the question is also how to show to the user if there is a difference between what he has enabled and what we found out.
> 
> Also a user may not want to see historical data for a protocol for some reason. For example I no longer care about DSR data. I want to disable it even though 2 of my accounts did participate. I want faster queries.
Yes I get this. It's unfortunate that we need to deal with this added complexity due to the query speed. Unfortunately as we add more protocols it will just become more cumbersome and slower. We should consider whether it's worth keeping this as a fallback and just using something like The Graph in the meantime. Introducing this complexity and in general the slowness is quite unattractive when compared to how fast some of the defi aggregators are. The downside (setting aside any issues with graph data accessibility in the long-run) is that every time you implement a protocol you will need to implement it in two ways (traditional contract queries & the graph) and then integrate them into the same backend data structure.

We also need to consider that "defi history" may include taxable events. What is the effect of disabling a defi module / specific account on incorporating defi history events in the tax report? If disabled == not included, then we encounter the issue of the user unknowingly omitting something from the tax report because they've disabled the module/account for faster queries.

Reply:
> We also need to consider that "defi history" may include taxable events. What is the effect of disabling a defi module / specific account on incorporating defi history events in the tax report? If disabled == not included, then we encounter the issue of the user unknowingly omitting something from the tax report because they've disabled the module/account for faster queries. If tax reports are unaffected by this, we should note it somehow/somewhere.

Disabling a module completely removes queries of that module. Tax report included. It's not bad. It's a feature. Not every user utilizes every protocol. If you don't query historical events for a protocol they won't be included in the tax report. This is something very simple to reason about.

>  We should consider whether it's worth keeping this as a fallback and just using something like The Graph in the meantime.

You assume that Graph is better or somehow faster. It's not. Plus it's specialized per protocol and not every protocol has a subgraph. Also queries will need to be made for each and every address the user tracks.

Querying anything, the local node, etherscan, infura, the graph whatever ... is REALLY expensive. Time-wise. It's many many RPC calls over a network. And with the graph it may even start costing per-query in the future. We simply can't have a system where every address the user tracks is queried for every possible protocol. That's not going to work or scale ever. Especially with ethereum where it's pretty standard to have many addresses, each for a different thing.

I think the current approach is fine, as long as the introductory text is also descriptive enough.

Additionally when/if we enhance the premium subscription to be per-module then the modules here would be enabled depending on the user's subscription.

Reply:
> Disabling a module completely removes queries of that module. Tax report included. It's not bad. It's a feature. Not every user utilizes every protocol. If you don't query historical events for a protocol they won't be included in the tax report. This is something very simple to reason about.

This is simple to you because you know the application inside and out. It is not simple to a user who doesn't. It's better to explain something than assume that users will reason about things in the same way as you.
We already have multiple issues open about things that are reasonable to you and me (and in some cases there are even warnings/explanations), but are _still_ causing usability problems, and what we're discussing now is far more convoluted than those.

 
> You assume that Graph is better or somehow faster. It's not. Plus it's specialized per protocol and not every protocol has a subgraph. Also queries will need to be made for each and every address the user tracks.

It's not faster? You're saying querying data from a sub-graph is just as slow as querying it via contracts on the mainnet? It's specifically built to index certain kinds of content and make retrieving them at large volumes quick and easy. The rest isn't something new -- every protocol you implement will have to be separately implemented, whether it's via specific contract calls or via a sub-graph (or some combination of the two). The last part about "separate queries for each address" I'm not sure I follow. Depending on the subgraph, you can definitely get info related to multiple addresses in one query.

Reply:
>  It's better to explain something 

Sure. Never said otherwise. As I said above, please suggest the contents of the text that you would prefer.

> You're saying querying data from a sub-graph is just as slow as querying it via contracts on the mainnet?

I have heard from others who have used the graph that it's not much faster. Especially with the way logs are indexed in places like etherscan. Plus they don't contain all the info we need so far for the protocols I checked.

Will have to look further into it and make our own experiments, but the introduction of the graph token has made me not want to devote too much time into it. Please understand that there is only one person on the backend at the moment. And I am also doing a lot of other things on the project so prioritization is important.

Anyway a review of this PR is not the place to have a discussion on this. If you want to discuss something graph related the issue is here: https://github.com/rotki/rotki/issues/640. We may experiment with it as a data source at some point.